### ISSUE TITLE
No confirmation when "h" key is used to create a highlight

### ISSUE THREAD

**Issue by mattdricker** [hypothesis/product-backlog#1426]
This feedback comes out of Accessibility Web Audit from UC Berkeley.

### Current behavior
When text is selected and the "h" key is used to create a highlight, the screen reader user does not get a confirmation that the highlight has been created.

### Expected behavior
Screen reader should announce confirmation when a highlight has been created.

This is the feedback from the review:

> H key to generate a highlight worked, but there was no spoken confirmation for the screen reader user.
>
> Screen reader users should hear a confirmation that text has been highlighted.

**Comment by acelaya** [comment]
I have been testing this in a couple of ways, and for me it works as expected.

Both when clicking on "Highlight" and when using the keyboard shortcut, the new highlight gets announced by the screen reader (I'm using orca).

There's one consideration thought. The aria-live region where we "append" messages to be notified, lives in the sidebar. If the sidebar is collapsed, the screen reader does not notify those messages, but this is again consistent both for keyboard shortcut and mouse click.

One thing worth mentioning, although I think it's not related, is that I get a weird behavior when pressing `h` key, only while the screen reader is running. In this case, it scrolls down when pressing `h`, and it does not create the highlight. This does not happen if the screen reader is not running 🤔 

I had to change the actual shortcut to a different key in order to test this, but as I said, I think it's not related, and this is probably something specific to orca.

**Comment by acelaya** [comment]
It was tricky, but I managed to reproduce the same behavior as in my previous comment, but with NVDA under Windows 10.

To summarize, the behavior is:

* If the sidebar is collapsed, highlights are not announced, regardless of the usage of mouse click, or `h` shortcut.
* While the sidebar is not collapsed, new highlights created with mouse click are properly announced.
* While the sidebar is not collapsed, new highlights created with `h` shortcut are properly announced.

When I say "it was tricky", it is because NVDA also seems to "steal" keyboard strokes while running, so I had to disable that behavior in order to test this.

**Comment by robertknight** [comment]
The immediate thought I have is that the person who reported the issue initially might not be aware of this behavior:

> If the sidebar is collapsed, highlights are not announced, regardless of the usage of mouse click, or h shortcut.

A sighted user gets feedback in this case because they see the highlight in the page, if not the sidebar. What would it take to change the behavior so that the annotation / highlight is always announced regardless of the sidebar state?

**Comment by mattdricker** [comment]
Good catch with the sidebar state affecting the announcement. Ideally I think we want to announce regardless of sidebar state, as non-sighted users aren't aware generally of whether the sidebar is collapsed or not at any given time.

**Comment by acelaya** [comment]
I guess we would need to change the `aria-live` region where we push those to make sure it gets notified. I will check what would be needed.

**Comment by acelaya** [comment]
I'm wondering. Considering the originally reported issue is actually not reproducible, but we have found another topic worth investigating, should we close this issue and open a new one?

**Comment by acelaya** [comment]
@mattdricker We have just released a new version of the client which make sure hidden notifications are announced by screen readers, even when the sidebar is collapsed.
